### PR TITLE
Update `ProposalProcedure` return address to be a `RewardAcnt`

### LINKED ISSUE #3594: Change `pProcReturnAddr` to `Credential`
Currently it is set to just a `KeyHash`, which is only 50% correct.

It [the spec](https://github.com/input-output-hk/formal-ledger-specifications/blob/e49fe43cafce0833d7ca8caf8016cd3e4d1467b2/src/Ledger/GovernanceActions.lagda#L129) however it is set to `RewardAcnt`, which would mean we also need to worry about the `NetworkId`. This does not seem right to me, so it would be nice to get some clarification on this

### PR DESCRIPTION
**Pull request by teodanciu**
# Description

Closes https://github.com/input-output-hk/cardano-ledger/issues/3594

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [ ] Self-reviewed the diff

### CODE REVIEW

**Comment by teodanciu** [comment]
I assume `scriptsNeeded` doesn't have to be updated for this, please correct me if I'm wrong

File: eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs [hunk 1]
```diff
@@ -98,7 +98,7 @@ data GovernanceActionState era = GovernanceActionState
   , gasDRepVotes :: !(Map (Credential 'DRepRole (EraCrypto era)) Vote)
   , gasStakePoolVotes :: !(Map (KeyHash 'StakePool (EraCrypto era)) Vote)
   , gasDeposit :: !Coin
-  , gasReturnAddr :: !(KeyHash 'Staking (EraCrypto era))
+  , gasReturnAddr :: !(Credential 'Staking (EraCrypto era))
```
Comment (lehins) [inline eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs]: Turns out it actually has to be a reward account, not just a `Credential`
```suggestion
  , gasReturnAddr :: !(RewardAcnt (EraCrypto era))
```

**Review by lehins — CHANGES_REQUESTED** [review]
During a meeting yesterday we figured out a discrepancy. This has to be a RewardAcnt, not just a Credential

**Comment by teodanciu** [comment]
I updated it to be a RewardAcnt, however, I haven't used the `Network` anywhere. 
I assume it will be needed (otherwise why have it), but not sure if at this point. Maybe in the way we insert in UMap in Epoch rule?

**Comment by lehins** [comment]
We'll only need to validate that the NetworkId matches the network we are running. That's all.

See this ticket I created: https://github.com/input-output-hk/cardano-ledger/issues/3603

File: eras/conway/impl/CHANGELOG.md [hunk 2]
```diff
@@ -50,6 +50,7 @@
 * Remove `DelegStakeTxCert` from the `COMPLETE` pragma for `TxCert`
 * Add `Committee` and adjust `NewCommittee` governance action
 * Add `ConwayUpdateDRep` constructor to `ConwayGovCert` type and corresponding pattern `UnRegDRepTxCert`
+* Update `ProposalProcedure` return address to be a `Credential`
```
Comment (lehins) [inline eras/conway/impl/CHANGELOG.md]:
```suggestion
* Update `ProposalProcedure` return address to be a `RewardAcnt`
```

**Review by lehins — APPROVED** [review]
Great. Thank you!

**Pull request merged**